app/src/main/java/com/google/maps/android/compose/MarkerClusteringActivity.kt

```diff
@@ -107,6 +108,181 @@ fun GoogleMapClustering(items: List<MyItem>) {
+@Composable
+private fun DefaultClustering(items: List<MyItem>) {
+    Clustering(
+        items = items,
+        // Optional: Handle clicks on clusters, cluster items, and cluster item info windows
+        onClusterClick = {
+            Log.d(TAG, "Cluster clicked! $it")
+            false
+        },
+        onClusterItemClick = {
+            Log.d(TAG, "Cluster item clicked! $it")
+            false
+        },
+        onClusterItemInfoWindowClick = {
+            Log.d(TAG, "Cluster item info window clicked! $it")
+        },
+        // Optional: Custom rendering for non-clustered items
+        clusterItemContent = null
+    )
+}
+
+@OptIn(MapsComposeExperimentalApi::class)
+@Composable
+private fun CustomUiClustering(items: List<MyItem>) {
+    Clustering(
+        items = items,
+        // Optional: Handle clicks on clusters, cluster items, and cluster item info windows
+        onClusterClick = {
+            Log.d(TAG, "Cluster clicked! $it")
+            false
+        },
+        onClusterItemClick = {
+            Log.d(TAG, "Cluster item clicked! $it")
+            false
+        },
+        onClusterItemInfoWindowClick = {
+            Log.d(TAG, "Cluster item info window clicked! $it")
+        },
+        // Optional: Custom rendering for clusters
+        clusterContent = { cluster ->
+            CircleContent(
+                modifier = Modifier.size(40.dp),
+                text = "%,d".format(cluster.size),
+                color = Color.Blue,
+            )
+        },
+        // Optional: Custom rendering for non-clustered items
+        clusterItemContent = null
+    )
+}
+
+@OptIn(MapsComposeExperimentalApi::class)
+@Composable
+fun CustomRendererClustering(items: List<MyItem>) {
+    val clusterManager = rememberClusterManager<MyItem>()
+    val renderer = rememberClusterRenderer(
+        clusterContent = { cluster ->
+            CircleContent(
+                modifier = Modifier.size(40.dp),
+                text = "%,d".format(cluster.size),
+                color = Color.Green,
+            )
+        },
+        clusterItemContent = {
+            CircleContent(
+                modifier = Modifier.size(20.dp),
+                text = "",
+                color = Color.Green,
+            )
+        },
+        clusterManager = clusterManager,
+    )
+    SideEffect {
+        clusterManager ?: return@SideEffect
+        clusterManager.setOnClusterClickListener {
+            Log.d(TAG, "Cluster clicked! $it")
+            false
+        }
+        clusterManager.setOnClusterItemClickListener {
+            Log.d(TAG, "Cluster item clicked! $it")
+            false
+        }
+        clusterManager.setOnClusterItemInfoWindowClickListener {
+            Log.d(TAG, "Cluster item info window clicked! $it")
+        }
+    }
+    SideEffect {
+        if (clusterManager?.renderer != renderer) {
+            clusterManager?.renderer = renderer ?: return@SideEffect
+        }
+    }
```

Comment:
For the high-level APIs, Processing is performed in Clustering Composable.
For the low-level API, I wondered what to do, but did not include it. Because I may be some inconvenience I'm not aware of.